### DISCUSSION TITLE
Request WMS legend with scale

### DISCUSSION THREAD

**Discussion by zoran995** [TerriaJS/terriajs#5090] [General]
WMS `GetLegendGraphic` supports returning different legend for different scale based on the rendering elements (section 10.2 of SLD specification). I know that calculating the scale is very problematic in 3D, but we can calculate an approximate scale value and send it with the request (also a trait so admin can set this).
It is happening quite often that user has different styling for different zoom levels and in those cases, legend contains a bunch of objects not visible on the map. What do others think about this?

**Comment by hardreddata** [comment]
This is an interesting idea. 

Thinking about the layers I look after on GeoServer the zoom based styling would really require the legend to be altered. That is, the layers are set to vanish when zoomed out, to conserve resources. If this was built I suppose I could use the legend to prompt the end user to zoom in.

**Comment by meh9** [comment]
What we have done with some stakeholders is have them produce a custom legend instead of using one autogenerated by Geoserver, but we don't change the legend at different zoom levels because when you view at an oblique angle you often see both styles at the same time, e.g.:

<img width="871" alt="Screen Shot 2021-01-13 at 11 50 55" src="https://user-images.githubusercontent.com/5283003/104392343-fbec7d80-5595-11eb-9af1-df2063ae5f43.png">

You almost want to request a legend for all the zoom levels that you are actually displaying, but again that opens up so many corner cases, and considering when you view something and include the horizon you're nearly guranteed to have _all_ zoom levels...

**Reply by meh9** [reply]
So yeah, that's not an answer, that's a "hmm this is complicated" :)
